src/goto-symex/equation_conversion_exceptions.h

```diff
@@ -0,0 +1,53 @@
+/*******************************************************************\
+
+Module: Symbolic Execution
+
+Author: Diffblue Ltd.
+
+\*******************************************************************/
+
+/// \file
+/// Exceptions that can be raised during the equation conversion phase
+
+#ifndef CPROVER_GOTO_SYMEX_EQUATION_CONVERSION_EXCEPTIONS_H
+#define CPROVER_GOTO_SYMEX_EQUATION_CONVERSION_EXCEPTIONS_H
+
+#include <ostream>
+#include <util/namespace.h>
+#include <langapi/language_util.h>
+#include "symex_target_equation.h"
+
+class guard_conversion_exceptiont : public std::runtime_error
+{
+public:
+  guard_conversion_exceptiont(
+    const symex_target_equationt::SSA_stept &step,
+    const namespacet &ns)
+    : runtime_error("Error converting guard for step"), step(step)
+  {
+    std::ostringstream error_msg;
+    error_msg << runtime_error::what();
+    error_msg << "\nSource GOTO statement: "
+              << from_expr(ns, "java", step.source.pc->code);
```

Comment:
Use `format(...)` from `util/format_expr.h` to avoid 1) the need for a namespace and 2) language-specifics when you don't actually have information about what language it is.